### PR TITLE
Account Recovery: Actions interacting with `/account-recovery/request-reset` endpoint

### PR DESCRIPTION
**Pull request by southp**
## Summary
This PR is based on #11558, aiming at adding action creators interacting with `/account-recovery/request-reset` endpoint.

## Test Plan

1. `npm run test-client client/state/account-recovery/reset/test`
1. `npm run test-client client/state/data-layer/wpcom/account-recovery/request-reset/test/`

### CODE REVIEW

File: client/state/action-types.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ export const ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR = 'ACCOUNT_RECOVERY_RESET_OPTI
 export const ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE = 'ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE';
 export const ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST = 'ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST';
 export const ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA = 'ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA';
+export const ACCOUNT_RECOVERY_RESET_REQUEST = 'ACCOUNT_RECOVERY_RESET_REQUEST';
```
Comment (omarjackman) [inline client/state/action-types.js]: Alphabetize :trollface: 

Lets keep this list of exports alphabetized 😄

File: client/state/account-recovery/reset/actions.js [hunk 2]
```diff
@@ -51,3 +54,25 @@ export const updatePasswordResetUserData = ( userData ) => ( {
...
+export const requestPasswordResetSuccess = () => ( {
+	type: ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
+} );
+
+export const requestPasswordResetError = ( error ) => ( {
+	type: ACCOUNT_RECOVERY_RESET_REQUEST_ERROR,
+	error,
+} );
+
+export const requestPasswordReset = ( request ) => ( dispatch ) => {
```
Comment (omarjackman) [inline client/state/account-recovery/reset/actions.js]: I think we're moving away from using action thunks and shifting towards doing this kind of thing in middleware. I wont be around much tonight to help out on that front but maybe @jordwest can give some pointers there?

I can take another stab at this tomorrow if needed

**Comment by southp** [comment]
@omarjackman Thanks for the reminder. It was in `thunk` simply because there is a issue that `wpcom-http` doesn't play well with `apiNamespace`, more details [here](https://github.com/Automattic/wp-calypso/pull/10878#issuecomment-282268167). However, it shouldn't block me from implementing it in the data layer since I can still use the plain old `wpcom.req.post` in this case.
I have rewritten it using the data layer, and it should be ready to be reviewed now :)

File: client/state/account-recovery/reset/actions.js [hunk 3]
```diff
@@ -3,9 +3,12 @@
...
 import wpcom from 'lib/wp';
 import {
-	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
-	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
 	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+	ACCOUNT_RECOVERY_RESET_REQUEST_RESET,
+	ACCOUNT_RECOVERY_RESET_REQUEST_RESET_SUCCESS,
+	ACCOUNT_RECOVERY_RESET_REQUEST_RESET_ERROR,
```
Comment (jordwest) [inline client/state/account-recovery/reset/actions.js]: Could these be:

```diff
-	ACCOUNT_RECOVERY_RESET_REQUEST_RESET,
-	ACCOUNT_RECOVERY_RESET_REQUEST_RESET_SUCCESS,
-	ACCOUNT_RECOVERY_RESET_REQUEST_RESET_ERROR,
+	ACCOUNT_RECOVERY_RESET_REQUEST,
+	ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
+	ACCOUNT_RECOVERY_RESET_REQUEST_ERROR,
```

Comment (southp) [inline client/state/account-recovery/reset/actions.js]: Ah, totally. I have no idea why they were there in the first place 😅
Fixed by 6d6d057

**Pull request merged**